_posts/byp/2021-10-20-byp-trinket-post.md

```diff
@@ -0,0 +1,23 @@
+---
+layout: post
+author: byp
+title: byp's trinket post!"
+---
+
+Here's an example about regex :
+
+<iframe src="https://trinket.io/embed/python/f3f9f9a9cf" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+
+```python
+import re
+s = ['X-DSPAM-Confidence: 0.8475', 'X-DSPAM-Probability: 0.0000', 'X-DSPAM-Confidence: 0.6178', 'X-DSPAM-Probability: 0.0000']
+for item in s:
+    x = re.findall('^X\S*: ([0-9.]+)', item)
+    if len(x) > 0:
+        print(x)
+```
+
+when you are using `findall()`, parentheses indicate that while you want the whole expression to match, you only are interested in extracting a portion of the substring that matches the regular expression.
```

Comment:
Yes, I think the parentheses is very powerful because it allows to filter to some texts one more step after doing the filter in general. And it might be very useful when doing the data extraction in data analysis. For example, I recently was involved in a Twitter user database analysis and with using the re library and this parenthesis, I'll be able to extract specific word I want to collect.